Makefile

```diff
@@ -219,6 +219,7 @@ generate-go-apis: ## Alias for .build/generate-go-apis
 
 .build/generate-go-apis: .build $(API_FILES) $(CONTROLLER_GEN) $(DEFAULTER_GEN) $(CONVERSION_GEN) ## Generate all Go api files
 	$(CONTROLLER_GEN) \
+		paths=./ \
```

Comment:
which go api we are building in root path?

Reply:
It is because I added the recommended  kubebuilder directives in `./main.go` for the new diagnostic endpoint following the deprecation of the `MetricsBindAddress` in controller-runtime. ([here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4739/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R120-R123))